### PR TITLE
Fix pacman cache removal in Arch image

### PR DESCRIPTION
**Pull request by Antiz96**
The `--noconfirm` pacman flag automatically accepts the default answer for each questions and the default answer to removing cached packages when using `pacman -Scc` is "N" (no).
That means that `pacman -Scc --noconfirm` does not actually remove cached packages.

Since pacman doesn't have a way to choose to answer yes or no automatically *(yet?)*, a simple workaround is to use `yes | pacman -Scc ` instead. However, while it is [recommended to set `pipefail` when using pipes](https://docs.docker.com/develop/develop-images/instructions/#using-pipes), it is not possible with the `yes` command (see [this comment](https://stackoverflow.com/a/70852402) for more details).  
Although, it should be safe to just use a piped `yes` without setting `pipefail` as I don't see how it could be a cause of failure in that particular case.

If using a pipe without setting `pipefail` is a concern, one can simply run `rm -f /var/cache/pacman/pkg/*` instead.  
I can adapt this PR accordingly if needed.

I remain available if needed :slightly_smiling_face:

### CODE REVIEW

**Review by Foxboron — APPROVED** [review]

**Comment by debarshiray** [comment]
@Antiz96 thanks for noticing this!

I wonder if it will be possible to add a test for this.  Maybe based on:
```
$ ls -1 /var/cache/pacman/pkg
```

In the past, I managed to hack in some tests for the `fedora-toolbox` images into their Container files.  It gives some assurance that future changes to the images or the underlying OS won't regress anything.

**Comment by Antiz96** [comment]
@debarshiray You're welcome :)

We could add a line like this for instance:

```
RUN [ -z "$(ls /var/cache/pacman/pkg)" ] || echo "non empty" # or exit, or whatever is needed
```

Although, to be honest, I'm not so worried about `yes | pacman -Scc` failing to clean the cache, but you can never be too careful I guess :smile:  
I can open another PR for the above test if you want to.

**Comment by debarshiray** [comment]
> @debarshiray You're welcome :)
> 
> Could add a line like this for instance:
> 
> ```
> RUN [ -z "$(ls /var/cache/pacman/pkg)" ] || echo "non empty" # or exit, or whatever is needed
> ```

Sounds good to me.
 
> Although, to be honest, I'm not so worried about `yes | pacman -Scc` failing to clean the cache, but you can never be too careful I guess 😄 I can open another PR for the above test if you want to.

Ultimately @Foxboron is the maintainer of the Arch Linux images.  So, I leave it to him.

I personally tend to err on the side of caution.  Someone might casually change something without checking, and then who knows?  After all, you did discover this problem where the existing attempt wasn't working as expected.  :)

**Pull request merged**
